src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -80,13 +81,6 @@ public AsyncLazy(T value)
             _cachedResult = Task.FromResult(value);
         }
 
-        /// <summary>
-        /// Important: callers of this constructor should ensure that the compute function returns
-        /// a task in a non-blocking fashion.  i.e. the function should *not* synchronously compute
-        /// a value and then return it using Task.FromResult.  Instead, it should return an actual
-        /// task that operates asynchronously.  If this function synchronously computes a value
-        /// then that will cause locks to be held in this type for excessive periods of time.
-        /// </summary>
         public AsyncLazy(Func<CancellationToken, Task<T>> asynchronousComputeFunction, bool cacheResult)
```

Comment:
It was wrong -- we haven't held the lock when calling compute functions for over a year now.